Source/WebCore/rendering/RenderObject.cpp

```diff
@@ -2417,6 +2417,18 @@ auto RenderObject::collectSelectionGeometriesInternal(const SimpleRange& range)
         }
     }
 
+    auto numberOfGeometries = geometries.size();
+
+    // If we have no geometries, do not even bother doing any more work.
+    if (!numberOfGeometries)
+        return { WTFMove(geometries), 0 };
```

Comment:
This part of the PR definitely makes sense (though "moving" empty vector is a bit odd and you can just do `return { }` and change
`int maxLineNumber` to
`int maxLineNumber { 0 }` (<- should have been done originally)
instead.

While the rest of the PR (removing handling the first line) looks correct, I am not convinced it is a forward looking change. However I am not the author/maintainer of this code so probably not the best person to review.

Reply:
Done!